src/hashfunctions.c

```diff
@@ -139,13 +139,16 @@ Obj DATA_HASH_FUNC_FOR_INT(Obj self, Obj i)
         return HashValueToObjInt(DataHashFuncForInt(i));
 }
 
-Int BasicRecursiveHash(Obj obj);
+static inline Int BasicRecursiveHash(Obj obj);
+
+// This is just a random number which fits in a 32-bit UInt.
+// It is used the base for hashing of lists
+#define LIST_BASE_HASH 2195952830L
```

Comment:
Nitpick, but: I tend to prefer enums over macros; this way the compiler sees the name.